### PR TITLE
remove all vendoring

### PR DESCRIPTION
**Pull request by jaypipes**
The `vendor/` directory makes mirror maintenance unnecessarily painful. Modern Go and modules are a cleaner, more maintainable way to manage dependencies.

This patch removes all use of Go vendoring and updates the Makefile targets to install dependent test tooling like `mockery` and `git-semver` into the `_out/` directory instead of looking in a `vendor/` directory.

Issue #293

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -140,13 +138,15 @@ _out/golangci-lint: outdir
 		echo "Using golangci-lint cached at $(GOLANGCI_LINT_BIN)";\
 	fi
 
-
 _out/git-semver: outdir
-	@go build -o _out/git-semver vendor/github.com/mdomke/git-semver/main.go
+	@GOBIN=$(shell pwd)/_out go install github.com/mdomke/git-semver/v6@latest
```
Comment (ffromani) [inline Makefile:142]: floating tags created a lot of pain in the past. Let's use `v6@v6.9.0` or any fixed, non-floating tag you prefer.

**Review by ffromani — COMMENTED** [review]
LGTM once we address the floating tag comment

File: Makefile (hunk 1 above)
Comment (jaypipes) [reply Makefile]: fixed! :)

**Review by ffromani — APPROVED** [review]
thanks, LGTM

**Pull request merged**